test/boost/memtable_test.cc

```diff
@@ -815,3 +815,67 @@ SEASTAR_TEST_CASE(failed_flush_prevents_writes) {
 #endif
 }
 
+SEASTAR_TEST_CASE(mutation_order_violation) {
```

Comment:
This doesn't belong in memtable_test any more. Better to produce the "bad" fragment stream directly, because going through a memtable risks it getting changed by the memtable compacting it away (see #10174